### DISCUSSION TITLE
Installing package outside fedora & rpmfusion repo

### DISCUSSION THREAD

**Discussion by xaolanx** [blue-build/.github#45] [Support]
Hello, I want to install sass but it's not in repo. I can install it using bun (not in repo too) or using npm. The question is can i make script or anything else to install sass during rebase?

**Comment by xynydev** [comment]
I think it's worth examining what you want to use `sass` for and if it's completely necessary in your image; it's something I would categorize as a software development tool and it can be easily installed locally.

There is `npm install --global`, however, that probably would install sass into `/usr/local/` which is a symlink to `/var/usrlocal/` and thus not supported. I guess one could copy the files out of there...

If you need `sass` during the build, but not in the final image, you could use a node/npm [build stage](https://blue-build.org/reference/stages/) to build whatever you want to build and copy only the result to your image.

`sass` is also distributed as a standalone binary; one option would be to make a script to download the correct [`.tar.gz` off of GitHub](https://github.com/sass/dart-sass/releases/tag/1.83.0) into `/tmp/` and then unpack that with `tar` and place the binary at `/usr/bin/sass`. This is the approach I'd recommend if you absolutely need `sass` installed in your image, and don't mind it not being managed by a package manager, and you having to manually update the package version number every time you want it to update (or alternatively having to make a script to always fetch the latest version). 

Personally, I would still just prefer running `npm install -g sass` or `brew install node-sass` locally and forgetting I even installed it manually, lol.
